docs/root/api/starting_envoy.rst

```diff
@@ -0,0 +1,74 @@
+.. _api_starting_envoy:
+
+Starting Envoy
+==============
+
+---------------
+``EnvoyClient``
+---------------
+
+Starting an instance of Envoy Mobile for making requests is done by creating an ``EnvoyClient``,
+which conforms to the ``HTTPClient`` interface.
+
+To do so, create an ``EnvoyClientBuilder`` and call ``build()``:
+
+**Kotlin example**::
+
+  val envoy = AndroidEnvoyClientBuilder(baseContext, Domain("api.envoyproxy.io"))
+    .addLogLevel(LogLevel.WARN)
+    .addStatsFlushSeconds(60)
+    ...
+    .build()
+
+**Swift example**::
+
+  let envoy = try EnvoyClientBuilder(domain: "api.envoyproxy.io")
+    .addLogLevel(.warn)
+    .addStatsFlushSeconds(60)
+    ...
+    .build()
+
+After the client is created, it should be stored and kept in memory in order to be used
+for issuing requests.
+
+----------------------
+Advanced configuration
+----------------------
+
+In most cases, the functions provided by the builder should cover basic setup requirements.
+However, in some cases it can be useful to provide a
+`Envoy configuration YAML file <https://www.envoyproxy.io/docs/envoy/latest/configuration/configuration>`_
+with additional customizations applied.
+
+This may be done by initializing a builder with the contents of the YAML file you you wish to use:
+
+**Kotlin example**::
+
+  val envoy = AndroidEnvoyClientBuilder(baseContext, Yaml(yamlFileString))
```

Comment:
Since providing a custom yaml string is not standard usage, I'd suggest we maybe omit setting the arguments in these examples.

Reply:
I figured it was worth documenting here since it's in the public API. There's a red `warning` banner below these examples advising against using it in most cases